scripts/test-utils/swap-lwc-style-for-style.ts

```diff
@@ -0,0 +1,100 @@
+/**
+ * SSRv2 implements style deduplication, where an <lwc-style> tag is inserted alongside the first instance
+ * of a given component's <style> tag. In subsequent renders of that component, only the <lwc-style> tag
+ * is present in the generated markup.
+ *
+ * This presents a problem when sharing fixtures between SSRv1 and SSRv2. SSRv1 will expect to find repeated
+ * <style> tags and won't expect to find <lwc-style> tags.
+ *
+ * The `swapLwcStyleForStyleTag` function takes the SSRv2 fixtures and modifies it to look how SSRv1 would
+ * expect. As more divergences between the implementations crop up, we may need to add more fixture mutations
+ * to accommodate the differences.
+ */
+
+interface StyleTagInfo {
+    originalStyleTag: string;
+    styleTagWithoutId: string;
+    idAttrStr: string;
+    styleId: string;
+    styleTagIdx: number;
+    indentation: number | null;
+}
+
+export function swapLwcStyleForStyleTag(src: string): string {
+    let modifiedSrc = src;
+    const styleCapture = /(\s*)(<style [^>]*(id="([^"]+)" ?)[^>]*>.*?<\/style>)/s;
+    const capturedStyleTags: StyleTagInfo[] = [];
+    let styleTagMatch: RegExpExecArray | null = null;
+
+    // Find all <style> tags, remove id attrs, capture information for later replacement of <lwc-style> tags
+    while ((styleTagMatch = styleCapture.exec(modifiedSrc))) {
```

Comment:
```suggestion
    for (const styleTagMatch of modifiesSrc.matchAll(styleCapture)) {
```
inline assignment is a gross pattern

Reply:
The suggested replacement won't work because the length of the string will be changing as `<style>` tags are modified. We really do want to `exec` each time. Alternately, we could queue up all the modifications and iterate over them in reverse after the fact.

I will add a comment explaining, and we can have a follow-up discussion about how to express this more cleanly.

Reply:
that's double gross lol

but nbd it's a test util